snapshotter/devmapper/snapshotter.go

```diff
@@ -186,7 +219,22 @@ func (s *Snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 	log.G(ctx).WithFields(logrus.Fields{"name": name, "key": key}).Debug("commit")
 
 	return s.withTransaction(ctx, true, func(ctx context.Context) error {
-		_, err := storage.CommitActive(ctx, key, name, snapshots.Usage{}, opts...)
+		id, _, _, err := storage.GetInfo(ctx, key)
+		if err != nil {
+			return err
+		}
+
+		deviceName := s.getDeviceName(id)
+		size, err := s.pool.GetUsage(deviceName)
```

Comment:
This is a `dmsetup` disk-based operation happening inside a transaction.  Can you split this into two separate transactions instead so that we don't have anything calling out into devicemapper in the middle of a transaction?

Reply:
`dmsetup status` is not going to be heavy disk operation. Essentially it ends up with reading the corresponding value from btree leaf ([1](https://github.com/torvalds/linux/blob/cffd425b90147296c627dfb13f7ea8f361d962b2/drivers/md/dm-thin-metadata.c#L1825), [2](https://github.com/torvalds/linux/blob/6cdc577a18a616c331f57e268c97466171cfc45f/drivers/md/dm-thin.c#L4387), [3](https://github.com/torvalds/linux/blob/6cdc577a18a616c331f57e268c97466171cfc45f/drivers/md/dm-init.c#L264), [4](https://github.com/torvalds/linux/blob/6cdc577a18a616c331f57e268c97466171cfc45f/drivers/md/dm-thin.c#L53)). I don't think it worth to optimize here, we'll get more noise than performance gain.

Reply:
Let's keep an eye on it going forward.  I'm concerned about the combination of the lock plus skipping `--noflush`, but we can move forward and keep this in mind if we experience issues in the future.